import/src/main/java/com/google/cloud/healthcare/imaging/dicomadapter/CStoreService.java

```diff
@@ -124,54 +155,87 @@ private void validateParam(String value, String name) throws DicomServiceExcepti
     }
   }
 
-  private void redactAndStow(Executor underlyingExecutor, InputStream inputStream,
-      IDicomWebClient dicomWebClient) throws Throwable {
-    ExecutorCompletionService<Void> ecs = new ExecutorCompletionService<>(underlyingExecutor);
+  private IDicomWebClient selectDestinationClient(String callingAet, Attributes attrs){
+    for(DestinationFilter filter: destinationMap.keySet()){
+      if(filter.matches(callingAet, attrs)){
+        return destinationMap.get(filter);
+      }
+    }
+    return defaultDicomWebClient;
+  }
 
-    PipedOutputStream pdvPipeOut = new PipedOutputStream();
-    PipedInputStream pdvPipeIn = new PipedInputStream(pdvPipeOut);
-    PipedOutputStream redactedPipeOut = new PipedOutputStream();
-    PipedInputStream redactedPipeIn = new PipedInputStream(redactedPipeOut);
+  private void processStream(Executor underlyingExecutor, InputStream inputStream,
+      List<StreamProcessor> processorList) throws Throwable {
+    if (processorList.size() == 1) {
```

Comment:
Do we have any test coverage for this block? Maybe we should add a section to the integration test where we initialize the adapter without redaction / transcoding to verify the basic case still works?

Reply:
Almost all unit tests use this path (except transcoding test). There is no complex initialization that would make it more likely to fail in integration vs unit test either.
Integration test is already quite overcomplicated too...